odl/operator/pspace_ops.py

```diff
@@ -26,8 +26,8 @@
 import numpy as np
 import scipy as sp
 
-from odl.operator.operator import Operator
-from odl.space.pspace import ProductSpace
+from odl.operator import Operator
```

Comment:
This seems a bit fragile to me. Doesn't it suffice that the imports in `operator/__init__.py` get out of order to get a circular dependency here? I haven't really experimented with this, but it seems a bit more robust to go all the way down *in the same subpackage*. In `default_ops.py` for example, you did not change the import path of `Operator`.

I'm not thinking so much of the core packages here, but rather about somebody who wants to add a module or subpackage. It is perhaps easier to explain that imports from the same subpackage should be with full path, while imports from a different subpackage can rely on this "import to top" feature.

Speaking of users: I think you get this spurious `could not import diagnostics` import error when there's a circular dependency, right? In that case, a FAQ entry on that topic would be great. And a short section in the developer docs about "How to import" would probably also be helpful.